python/cuml/_thirdparty/sklearn/preprocessing/_data.py

```diff
@@ -835,14 +850,15 @@ def inverse_transform(self, X, copy=None):
                 X *= self.scale_
             if self.with_mean:
                 X += self.mean_
-
-        X = to_output_type(X, output_type)
         return X
 
     def _more_tags(self):
-        return {'allow_nan': True}
+        return {'X_types_gpu': ['2darray', 'sparse'],
```

Comment:
@viclafargue  we're in the process of making the tags system static in #3257, so depending on timing that PR will affect this one or the other way around. Do you foresee many issues arising from that change for these classes in `_thirdparty/sklearn`?

Reply:
No problem, I'll wait for your PR. Should be fairly simple, the models inherit from the `Base` class. I'll just have to make `_more_tags` methods static everywhere.

Reply:
@viclafargue I think were interested in knowing if you have any tags that are dynamic and will change from one instance to the other depending on the properties of the class. Or can all of the tags be determined in a static method.

Reply:
@dantegd Look at these lines: https://github.com/rapidsai/cuml/pull/3270/files#diff-6eb3baa332b96d9c2a1a79101b54e7d74c7510d65407a7cb54db9d3606d406f6R140-R141

Reply:
Sorry, I didn't knew these tags could be instance specific. From what I could see, all of them seems to be class specific (static) for preprocessing. I'll have to double check if a call to `fit` with sparse input wouldn't cause a model to only accept sparse transformations though.

Reply:
which tag would be instance specific?

Reply:
ah I see it, let me think on on it for a second, have a couple of ideas

Reply:
Since this is using the `AllowNaNTagMixin` already, instead of defining _more_tags, you can just add https://github.com/rapidsai/cuml/blob/c4c406883735ac4156c6b32005618b96ebeefda3/python/cuml/common/mixins.py#L306